js_src/header.ts

```diff
@@ -0,0 +1,73 @@
+export default class Header {
+    private refersToElement;
+
+    constructor(public element) {
+        HeaderSlideToggle.find(element, "[data-header-slide]");
+    }
+    
+    static apply() {
+        const elements = document.querySelectorAll(".header");
+        let toggles = [];
+        for (let i = 0; i < elements.length; i++) {
+            toggles.push(new Header(elements[i]));
+        }
+        return toggles;
+    }
+};
+
+export class HeaderSlideToggle {
+    private refersToElement;
+
+    public static softDismissDuration = 250; // ms
+
+    constructor(public element) {
+        this.refersToElement = document.querySelector(element.getAttribute("data-header-slide"));
+        
+        element.addEventListener("click", (e) => { this.toggle(); e.preventDefault(); });
+    }
+
+    private toggle() {
+        this.refersToElement.classList.toggle("is-active");
+        this.element.classList.toggle("is-active");
+
+        console.log(this.refersToElement)
+
+        // Position header slide appropriately relative to
+        // trigger.
+        // If the item is more on the left: left-align, if
+        // it is more on the right: right-align.
+        const rect = this.element.getBoundingClientRect();
+        this.refersToElement.style.top = (window.scrollY + rect.top + rect.height) + "px";
+
+        // reset max width, left and right
+        this.refersToElement.style.maxWidth = "";
+        this.refersToElement.style.left = "";
+        this.refersToElement.style.right = "";
+
+        const isRight = (rect.left + rect.width / 2) > document.body.clientWidth / 2;
+        // ^ check, whether the trigger is more on the left or on the right
```

Comment:
If this comment refers to the line above it, move the comment to the line above the relevant statement and drop the `^`